viewer/packages/tools/src/Measurement/MeasurementDistance.ts

```diff
@@ -0,0 +1,179 @@
+    this._positions = new Float32Array(6);
+    this._lineOptions = lineOptions ?? this._lineOptions;
+    this._raycaster = new THREE.Raycaster();
+    this._cameraDistance = 0;
+  }
+
+  private initializeLine(): void {
+    if (this._measurementLine === undefined) {
+      this._lineGeometry = new LineGeometry();
+      const lineMaterial = new THREE.RawShaderMaterial({
+        uniforms: THREE.UniformsUtils.clone(this.ShaderUniforms.uniforms),
+        vertexShader: lineShaders.vertex,
+        fragmentShader: lineShaders.fragment,
+        clipping: true,
+        glslVersion: THREE.GLSL3
+      });
+
+      this._positions[0] = this._positions[1] = this._positions[2] = 0;
+      this._positions[3] = this._positions[4] = this._positions[5] = 0;
+
+      this._lineGeometry.setPositions(this._positions);
+
+      this._measurementLine = new Line(this._lineGeometry, lineMaterial);
+
+      this._viewer.addObject3D(this._measurementLine);
+    }
+  }
+
+  /**
+   * Add/Start the line of the distance measurement
+   * @param point Start point of the Line
+   */
+  add(point: THREE.Vector3): void {
+    this.initializeLine();
+    this.addStartPoint(point);
+  }
+
+  private clearLine(): void {
+    if (this._measurementLine) {
+      this._measurementLine.geometry.dispose();
+      this._measurementLine.clear();
+      this._measurementLine = undefined;
+      this._isActive = false;
+    }
+  }
+
+  /**
+   * Remove the measurement line
+   */
+  remove(): void {
+    this.clearLine();
+  }
+
+  /**
+   * Completes a line for distance measurement
+   */
+  complete(): void {
+    this.clearLine();
+  }
+
+  private addStartPoint(point: THREE.Vector3): void {
+    this._startPoint.copy(point);
+    this._endPoint.copy(point);
+    this._positions[0] = this._startPoint.x;
+    this._positions[1] = this._startPoint.y;
+    this._positions[2] = this._startPoint.z;
+    this._isActive = true;
+  }
+
+  /**
+   * Get the distance value between the start point & end/mouse point
+   * @returns Distance between the two points
+   */
+  getMeasurementValue(): number {
+    return this._endPoint.distanceTo(this._startPoint);
+  }
+
+  /**
+   * Update the line
+   * @param x Screen X Position
+   * @param y Screen Y Position
+   */
+  update(x: number, y: number, endPoint?: THREE.Vector3): void {
+    const position = new THREE.Vector3();
+    if (endPoint) {
+      position.copy(endPoint);
+    } else {
+      const mouse = new THREE.Vector2();
+      mouse.x = (x / this._viewer.domElement.clientWidth) * 2 - 1;
+      mouse.y = -(y / this._viewer.domElement.clientHeight) * 2 + 1;
+      this._raycaster.setFromCamera(mouse, this._viewer.getCamera());
+
+      this._raycaster.ray.at(this._cameraDistance, position);
+    }
+
+    this._positions[3] = position.x;
+    this._positions[4] = position.y;
+    this._positions[5] = position.z;
+    this._lineGeometry.setPositions(this._positions);
+    this._endPoint.copy(position);
```

Comment:
This should clearly indicate that you compute a point at the same distance as the first point relative to the camera.